Directory.Solution.targets

```diff
@@ -0,0 +1,79 @@
+<Project>
+    <Target Name="RestoreTools">
+        <Exec Command="dotnet tool restore" />
+    </Target>
+
+    <Target
+            Name="DiscoverSourceFiles"
+            Outputs="FormatSources"
+            Returns="FormatSources">
+        <ItemGroup>
+            <FormatSources
+                    Include="src/**/*.fs"
+                    Exclude="src/**/obj/**/*.fs;src/Fantomas.FCS/generated/netstandard2.0/*.fs" />
+            <FormatSources Include="src/**/*.fsi" Exclude="src/Fantomas.FCS/generated/netstandard2.0/*.fsi" />
+            <FormatSources Include="build.fsx" />
+        </ItemGroup>
+    </Target>
+
+    <Target
+            Name="Format"
+            DependsOnTargets="RestoreTools;DiscoverSourceFiles">
+        <Exec Command="dotnet fantomas @(FormatSources, ' ')" />
+    </Target>
+
+    <Target
+            Name="CheckFormat"
+            DependsOnTargets="RestoreTools;DiscoverSourceFiles">
+        <Exec Command="dotnet fantomas --check @(FormatSources, ' ')" />
+    </Target>
+
+    <Target
+            Name="_GetChangedFilesInWorkingCopy"
+            Outputs="ChangedFiles"
+            Returns="ChangedFiles">
+        <Exec
+                Command="git diff --name-only"
+                ConsoleToMsBuild="true"
+                EchoOff="true">
+            <Output
+                    TaskParameter="ConsoleOutput"
+                    ItemName="ChangedFiles" />
+        </Exec>
+
+    </Target>
+
+    <Target
+            Name="FormatChanged"
+            DependsOnTargets="RestoreTools;_GetChangedFilesInWorkingCopy">
+        <ItemGroup>
+            <SourceFiles
+                    Include="%(ChangedFiles.Identity)"
+                    Condition="@(ChangedFiles->StartsWith('src'))" />
+            <FormatItems
+                    Include="@(SourceFiles)"
+                    Condition="'%(SourceFiles.Extension)' == '.fs' or '%(SourceFiles.Extension)' == '.fsi'" />
+        </ItemGroup>
+        <Exec
+                Command="dotnet fantomas @(FormatItems, ' ')"
+                Condition="@(FormatItems->Count()) > 0" />
+    </Target>
+
+    <Target Name="EnsureRepoConfig">
+        <Exec Command="git config core.hooksPath .githooks" />
+    </Target>
+
+    <Target Name="Benchmark">
+        <Exec Command="dotnet build -c Release src/Fantomas.Benchmarks" />
+        <Exec Command="dotnet src/Fantomas.Benchmarks/bin/Release/net6.0/Fantomas.Benchmarks.dll" IgnoreStandardErrorWarningFormat="true" />
+    </Target>
+    
+    <Target Name="All">
```

Comment:
How acceptable does this look like?
It is what we do today with FAKE but so much more straightforward 😱

Reply:
I got this working pretty well locally - calling targets on the solution is done through the `MSBuild` task.